lib/takeaway.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'order'
+require_relative 'menu'
+require_relative 'message_handler'
+
+class Takeaway
+  attr_accessor :menu_obj, :order_obj, :main_menu, :current_order, :message_handler
+  def initialize (menuClass, orderClass, message_handlerClass)
+    @menu_obj = menuClass.new
+    @order_obj = orderClass.new
+    @message_handler = message_handlerClass.new
+
+    @main_menu = menu_obj.menu
+    @current_order = order_obj.orders
+  end
+
+  def show_menu
+    menu_obj.format_menu
+  end
+
+  def add_to_main_menu item, price
+    menu_obj.add_dish item, price
+  end
+
+  def take_order item, quantity = 1
+    quantity.times do
+      current_order << main_menu.select {|dish, price| dish == item }
+    end
+  end
+
+  def check_order
+    puts "You have ordered..."
+    item_num = 1
+    current_order.each do |dish, price|
+      dish.each do |k,v|
+        puts "Item#{item_num}: #{k} that costs £#{v}"
```

Comment:
Remember, your user is probably not going to be using terminal. When you use "puts" your tying the presentation of your system to terminal. Imagine we wanted this text to instead be printed on a web page. In that case it would suffice to have the text returned as a single string or something.